src/styles/main.scss

```diff
@@ -1,3 +1,59 @@
-body {
-  margin: 0;
+@import "./utils/_variables";
+@import "./utils/normalize";
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  padding: 10px;
+  width: calc(7 * #{$cell-size } + 18px);
+  color: #000;
+
+  &__cell {
+    display: flex;
+    justify-content: center;
+    box-sizing: border-box;
+    height: $cell-size;
+    width: $cell-size;
+    background-color: #eee;
+    color: #000;
+    border: 1px solid black;
+    cursor: pointer;
+    transition: color 0.5s, transform 0.5s;
+  
+
+    &:hover {
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+  
+    @for $index from 1 through 31 {
+      &_:nth-child(#{$index})::before {
+        content: "#{$index}";
+      }
+    }
+  }
+
+  @for $index from 28 through 31 {
+    &--month-length--#{$index} &__cell:nth-last-child(-n + #{31 - $index}) {
+      display: none;
+    }
+  }
+
+  $start-days:
+    "mon" 1px,
+    "tue" 103px,
```

Comment:
instead of these magic numbers you can calculate these with variables 
use days of week number here like 
$days:
```
  "mon" 0,
  "tue" 1,
  "wed" 2,
  "thu" 3,
  "fri" 4,
  "sat" 5,
  "sun" 6;
```

Reply:
done.